src/main/clojure/clara/rules.cljc

```diff
@@ -332,6 +332,8 @@
       * :forms-per-eval - The maximum number of expressions that will be evaluated per call to eval.
         Larger batch sizes should see better performance compared to smaller batch sizes. (Only applicable to Clojure)
         Defaults to 5000, see clara.rules.compiler/forms-per-eval-default for more information.
+      * :retain-compile-ctx - Retain compilation information for better compilation failures.
+        Defaults to false, see clara.rules.compiler/retain-compile-ctx-default for more information.
 
       This is not supported in ClojureScript, since it requires eval to dynamically build a session. ClojureScript
       users must use pre-defined rule sessions using defsession."
```

Comment:
What happens when a session is deserialized without the compile-ctx and there is an error?  Does it revert gracefully to previous behavior?  This seems like something that might merit a test.

Reply:
Added a test to the test-durability ns with examples of the message that should be displayed.